### PR TITLE
Add get rosa-versions function

### PR DESCRIPTION
**Pull request by chetna14manku**

### CODE REVIEW

**Comment by redhat-qe-bot2** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: rosa/rosa_versions.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import rosa.cli
+from ocm_python_wrapper.ocm_client import OCMPythonClient
+
+
+def get_rosa_versions(
+    ocm_client: OCMPythonClient, aws_region: str, channel_group: str = "stable", hosted_cp: bool = False
+) -> Dict[str, Dict[str, List[str]]]:
+    """
+    Get all rosa versions for specified channel group.
```
Comment (myakove) [inline rosa/rosa_versions.py:12]: Please add empty line before `Args:`

File: rosa/rosa_versions.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        dict: ROSA versions
+
+    Examples:
+        >>> get_rosa_versions(ocm_client=ocm_client, aws_region="us-east-1", channel_group='candidate')
+        {'candidate': {'4.17': ['4.17.0-rc.0', '4.17.0-ec.3', '4.17.0-ec.2', '4.17.0-ec.1', '4.17.0-ec.0'],
+                       '4.16': [4.16.3', '4.16.2', '4.16.1', '4.16.0', '4.16.0-rc.9', '4.16.0-rc.6'],
+                       '4.15': ['4.15.32', '4.15.31', '4.15.30', '4.15.29', '4.15.28', '4.15.27']}}
+
+    """
+    rosa_base_available_versions_dict = {}
```
Comment (myakove) [inline rosa/rosa_versions.py:29]: Please set typing

File: rosa/rosa_versions.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    base_available_versions = rosa.cli.execute(
+        command=(f"list versions --channel-group={channel_group} {'--hosted-cp' if hosted_cp else ''}"),
+        aws_region=aws_region,
+        ocm_client=ocm_client,
+    )["out"]
+    _all_versions = [ver["raw_id"] for ver in base_available_versions]
```
Comment (myakove) [inline rosa/rosa_versions.py:35]: ditto

**Review by myakove — CHANGES_REQUESTED** [review]

**Review by myakove — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Pull request merged**